nowcasting_dataset/data_sources/gsp/gsp_data_source.py

```diff
@@ -399,14 +412,53 @@ def drop_gsp_by_threshold(gsp_power: pd.DataFrame, meta_data: pd.DataFrame, thre
 
     keep_index = maximum_gsp > threshold_mw
 
-    logger.debug(f"Dropping {sum(~keep_index)} GSPs as maximum is not greater {threshold_mw} MW")
+    dropped_gsp_names = meta_data.loc[keep_index[~keep_index].index].gsp_name
+
+    logger.debug(
+        f"Dropping {sum(~keep_index)} GSPs as maximum is not greater {threshold_mw} MW:"
+        f" Dropped GSP IDs and GSP names: {dropped_gsp_names}"
+    )
     logger.debug(f"Keeping {sum(keep_index)} GSPs as maximum is greater {threshold_mw} MW")
 
-    gsp_power = gsp_power[keep_index.index]
-    gsp_ids = gsp_power.columns
-    meta_data = meta_data[meta_data["gsp_id"].isin(gsp_ids)]
+    filtered_gsp_power = gsp_power.loc[:, keep_index]
```

Comment:
`gsp_power[keep_index.index]` doesn't modify `gsp_power` :slightly_smiling_face:

Reply:
yea - think its because the gsp's are in columns.